libs/remote_storage/src/error.rs

```diff
@@ -15,6 +17,9 @@ pub enum DownloadError {
     ///
     /// Concurrency control is not timed within timeout.
     Timeout,
+    /// Some integrity/consistency check failed during download. This is used during
+    /// timeline loads to cancel the load of a tenant if some timeline detects fatal corruption.
+    Fatal(Cow<'static, str>),
```

Comment:
nit: I think we can just use a string instead of a `Cow` here. Do we really care about allocations on error paths?

Reply:
Yeah.  This is a habit from places where we would use static strings sometimes, but in this case it's not useful.

42e943ed1bdbb383e3b8097f12eb0b6fe0fec31d